src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -681,9 +711,15 @@ export class SearchSource {
    * flat representation (taking into account merging rules)
    * @resolved {Object|null} - the flat data of the SearchSource
    */
-  private mergeProps(root = this, searchRequest: SearchRequest = { body: {} }): SearchRequest {
+  private mergeProps(
+    root = this,
+    searchRequest: SearchRequest = { body: {} },
+    filter?: string[]
+  ): SearchRequest {
```

Comment:
I have to admit that I don't really understand what these mergeProps changes are doing.

Reply:
I also didn't 
but now


What do I do with it? It's needed in `loadDataViewFields`
```
    const request = this.mergeProps(this, { body: {} }, ['query', 'filter']);
```

for getting fields to fetch I need merged props, if mergeProps without filtering down to those needed, it runs into code that needs fields.


 If I just merge query & filter, I get what's needed for the field retrieval.